File: sdk/ts/src/reason_codes.test.ts
import { describe, expect, it } from "vitest";
import { ReasonCodes, isRegisteredReasonCode } from "./index.js";

describe("reason codes", () => {
  it("names every registry code as its wire string", () => {
    expect(Object.keys(ReasonCodes)).toHaveLength(104);
    expect(ReasonCodes.EMERGENCY_STOP_FENCED).toBe("EMERGENCY_STOP_FENCED");
    expect(isRegisteredReasonCode("EMERGENCY_STOP_FENCED")).toBe(true);
    expect(isRegisteredReasonCode("NOT_A_REGISTERED_CODE")).toBe(false);
  });
});

File: sdk/go/reasoncodes/reasoncodes_test.go
package reasoncodes

import "testing"

func TestRegistryStrings(t *testing.T) {
	if len(All()) != 110 {
		t.Fatalf("All() = %d codes, want the registry's 110", len(All()))
	}
	if !IsRegistered(EmergencyStopFenced) || EmergencyStopFenced != "EMERGENCY_STOP_FENCED" {
		t.Fatal("EMERGENCY_STOP_FENCED must be registered under its wire string")
	}
	if IsRegistered("NOT_A_REGISTERED_CODE") {
		t.Fatal("an unknown code must classify as unregistered, not match")
	}
}
